Lib/filecmp.py

```diff
@@ -138,10 +138,20 @@ def __init__(self, a, b, ignore=None, hide=None, *, shallow=True): # Initialize
         self.shallow = shallow
 
     def phase0(self): # Compare everything except common subdirectories
-        self.left_list = _filter(os.listdir(self.left),
-                                 self.hide+self.ignore)
-        self.right_list = _filter(os.listdir(self.right),
-                                  self.hide+self.ignore)
+        try:
+            left_full_list = os.listdir(self.left)
+        except (OSError, ValueError):
+            self.left_list = []
```

Comment:
Is it right to silence errors here? Currently an exception is raised for non-accessible directories.

Reply:
It's always good when we receive feedback... After a more careful thinking, it does not make sense what I did. While `phase2` and `phase3` handle `OSError` and `ValueError`, it's because they also post-process those errors. Actually, I shouldn't have touched `dircmp` at all because `os.stat()` inside `phase2` can raise `ValueError` only if an invalid path is present in `self.common`, but this can only happen if `os.listdir` succeeded in the first place, which can only happen if when calling `phase0`, the contents of `self.left` are actually legit paths.

In summary:

- `self.common_funny` requires to compute `self.common`
- `self.common` requires to compute `self.left_list`
- `self.left_list` requires to call `os.listdir(self.left)`

So, if `os.listdir()` fails due to a ValueError (not an OSError), then I will never be able to construct `self.common`. And thus, I will also never be able to call `os.stat()` when constructing `self.common_funny`. It might happen that I have an OSError when a file disappears but not a ValueError due to the characters since those bad filenames wouldn't be in `self.common` in the first place.

Reply:
I'll let the protection against ValueError when doing `os.stat()` in `phase2` just in case a subclass does not handle it or does something funny.